File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/molecules/placeholder/MolCardPlaceholder.kt
package com.perrystreet.woof.designsystem.atomic.molecules.placeholder

import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.perrystreet.woof.designsystem.atomic.atoms.placeholder.AtomRoundedRectanglePlaceholder
import com.perrystreet.woof.designsystem.atomic.atoms.placeholder.atomPlaceholderShimmer
import com.perrystreet.woof.designsystem.theme.Theme

@Composable
fun MolCardPlaceholder() {
    AtomRoundedRectanglePlaceholder(
        modifier = Modifier
            .fillMaxWidth()
            .aspectRatio(Theme.aspectRatios.gridCell)
            .atomPlaceholderShimmer(),
    )
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Molecules/Placeholder/MolCardPlaceholder.swift
import SwiftUI

public struct MolCardPlaceholder: View {
    @Environment(\.theme) private var theme

    public init() {}

    public var body: some View {
        AtomRoundedRectanglePlaceholder()
            .aspectRatio(theme.aspectRatios.gridCell, contentMode: .fit)
            .atomPlaceholderShimmer()
    }
}
